### ISSUE TITLE
`spack load` broken by #22588

### ISSUE THREAD

**Issue by adamjstewart** [LLNL/spack#23701]
### Before

```console
$ git checkout cefbe48c89209dc3df654795644973b1885cdea4~1
$ spack load --first --sh python
# modifies the following variables:
SPACK_LOADED_HASHES
CPATH
MANPATH
PKG_CONFIG_PATH
PATH
ACLOCAL_PATH
DYLD_FALLBACK_LIBRARY_PATH
CMAKE_PREFIX_PATH
```

### After

```console
$ git checkout cefbe48c89209dc3df654795644973b1885cdea4
$ spack load --first --sh python
# modifies the following environment variables:
DYLD_FALLBACK_LIBRARY_PATH
CPATH
SPACK_LOADED_HASHES
```

My `modules.yaml` looks like:
```yaml
modules:
  enable:: []
```
but even without this file I can reproduce the bug. Any idea what could be causing this?

* **Spack:** 0.16.1-2651-fc392d2f56
* **Python:** 3.8.7
* **Platform:** darwin-catalina-ivybridge
* **Concretizer:** clingo

**Comment by adamjstewart** [comment]
For some reason this wasn't caught by our `cmd/load.py` tests. I'm guessing this only affects old installations, not new installations.

**Comment by eugeneswalker** [comment]
Good catch!

It looks like https://github.com/spack/spack/pull/22588 is also responsible for https://github.com/spack/spack/issues/23665 which uniquely affected RHEL 8 and not RHEL 7, Ubuntu 18.04, or Ubuntu 20.04.

**Comment by tgamblin** [comment]
This was fixed by #23674.
